### PR TITLE
Fix target _blank link icon alignment

### LINKED ISSUE #3048: CSS on Link (target=_blank) specificity issue
### Package name(s)

Core (@salt-ds/core)

### Package version(s)

    "@salt-ds/core": "^1.17.0",
    "@salt-ds/icons": "^1.9.1",
    "@salt-ds/theme": "^1.11.1",

### Description

The "external" icon is too close to the link, because .saltIcon is taking precedence over .saltLink-icon - and margin is overridden.

Github is not letting me upload images, but I have screenshots of the broken css and the chrome inspector - please contact me internally.

### Steps to reproduce

```
const Repro: React.FunctionComponent = () => {
    return (
        <div>
            <Link target="_blank" href={"whatever"}>
                Whatever
            </Link>
        </div>
    );
};
```

### Expected behavior

I expect the margins to behave like they do in the saltdesignsystem.com docs, but they do not.

### Operating system

- [ ] macOS
- [X] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

### Browser

- [X] Chrome
- [ ] Safari
- [ ] Firefox
- [ ] Edge

### Are you a JPMorgan Chase & Co. employee?

- [X] I am an employee of JPMorgan Chase & Co.

### PR DESCRIPTION
**Pull request by joshwooding**
Closes #3048

### CODE REVIEW

**Review by origami-z — COMMENTED** [review]
The link icon is lower than i expected, especially in LD/TD in the QA grid?

**Comment by joshwooding** [comment]
> The link icon is lower than i expected, especially in LD/TD in the QA grid?

Reverted the change, I did play around with a less hardcoded approach e.g. top: calc((var(--salt-text-lineHeight) - var(--icon-size) / 2). But it's quite ugly and for now -2px works across all densities anyway

File: packages/core/src/link/Link.css [hunk 1]
```diff
@@ -31,8 +31,8 @@
 }
 
 /* External link's icon */
-.saltLink-icon {
-  margin-left: var(--salt-size-adornmentGap);
```
Comment (origami-z) [inline packages/core/src/link/Link.css:35]: Spec need to change?

**Review by origami-z — APPROVED** [review]

**Review by lilyvc — APPROVED** [review]

File: packages/core/src/link/Link.css (hunk 1 above)
Comment (joshwooding) [reply packages/core/src/link/Link.css:35]: I don't think it's even in the spec

**Pull request merged**